pkg/reconciler/pubsub/reconciler.go

```diff
@@ -18,221 +18,145 @@ package pubsub
-		Secret:              secret,
-		Owner:               owner,
-	})
-}
-
-func (c *PubSubBase) EnsureSubscriptionDeleted(ctx context.Context, owner kmeta.OwnerRefable, secret corev1.SecretKeySelector, project, topic, subscription string) (ops.OpsJobStatus, error) {
-	return c.ensureSubscriptionJob(ctx, operations.SubArgs{
-		Image:          c.SubscriptionOpsImage,
-		Action:         ops.ActionDelete,
-		ProjectID:      project,
-		TopicID:        topic,
-		SubscriptionID: subscription,
-		Secret:         secret,
-		Owner:          owner,
-	})
-}
-
-func (c *PubSubBase) EnsureTopicExists(ctx context.Context, owner kmeta.OwnerRefable, secret corev1.SecretKeySelector, project, topic string) (ops.OpsJobStatus, error) {
-	return c.ensureTopicJob(ctx, operations.TopicArgs{
-		Image:     c.TopicOpsImage,
-		Action:    ops.ActionExists,
-		ProjectID: project,
-		TopicID:   topic,
-		Secret:    secret,
-		Owner:     owner,
-	})
-}
-
-func (c *PubSubBase) EnsureTopicCreated(ctx context.Context, owner kmeta.OwnerRefable, secret corev1.SecretKeySelector, project, topic string) (ops.OpsJobStatus, error) {
-	return c.ensureTopicJob(ctx, operations.TopicArgs{
-		Image:     c.TopicOpsImage,
-		Action:    ops.ActionCreate,
-		ProjectID: project,
-		TopicID:   topic,
-		Secret:    secret,
-		Owner:     owner,
-	})
-}
-
-func (c *PubSubBase) EnsureTopicDeleted(ctx context.Context, owner kmeta.OwnerRefable, secret corev1.SecretKeySelector, project, topic string) (ops.OpsJobStatus, error) {
-	return c.ensureTopicJob(ctx, operations.TopicArgs{
-		Image:     c.TopicOpsImage,
-		Action:    ops.ActionDelete,
-		ProjectID: project,
-		TopicID:   topic,
-		Secret:    secret,
-		Owner:     owner,
-	})
-}
-
-func (c *PubSubBase) ensureTopicJob(ctx context.Context, args operations.TopicArgs) (ops.OpsJobStatus, error) {
-	job, err := c.getJob(ctx, args.Owner.GetObjectMeta(), labels.SelectorFromSet(operations.TopicJobLabels(args.Owner, args.Action)))
-	// If the resource doesn't exist, we'll create it
-	if apierrs.IsNotFound(err) {
-		c.Logger.Debugw("Job not found, creating.")
+// ReconcilePubSub reconciles Topic / PullSubscription given a PubSubSpec.
+// Sets the following Conditions in the Status field appropriately:
+// "TopicReady", and "PullSubscriptionReady"
+// Also sets the following fields in the pubsubable.Status upon success
+// TopicID, ProjectID, and SinkURI
+func (psb *PubSubBase) ReconcilePubSub(ctx context.Context, pubsubable duck.PubSubable, topic, resourceGroup string) (*pubsubsourcev1alpha1.Topic, *pubsubsourcev1alpha1.PullSubscription, error) {
+	if pubsubable == nil {
+		return nil, nil, fmt.Errorf("nil pubsubable passed in")
+	}
+	namespace := pubsubable.GetObjectMeta().GetNamespace()
+	name := pubsubable.GetObjectMeta().GetName()
+	spec := pubsubable.PubSubSpec()
+	status := pubsubable.PubSubStatus()
 
-		job = operations.NewTopicOps(args)
+	topics := psb.pubsubClient.PubsubV1alpha1().Topics(namespace)
+	t, err := topics.Get(name, v1.GetOptions{})
```

Comment:
nit: seems no need to have two lines.

Reply:
not needed in the Delete. I just change it to have one line. Thanks!
Here is needed because afterwards it creates the topic with `topics`.